src/instructlab/training/main_ds.py

```diff
@@ -12,8 +12,18 @@
 
 # Third Party
 from accelerate import Accelerator
-from deepspeed.ops.adam import DeepSpeedCPUAdam, FusedAdam
-from deepspeed.runtime.zero.utils import ZeRORuntimeException
+try:
+    from deepspeed.ops.adam import DeepSpeedCPUAdam, FusedAdam
+except ImportError:
+    DeepSpeedCPUAdam = None
+    FusedAdam = None
+    print("DeepSpeed Adam optimizers are not available. Some features may be unavailable.")
```

Comment:
I like where you're headed with this. One thing we should do though is guard this print statement to the first rank to prevent it from appearing everywhere in a multi-GPU setting:

```suggestion
    local_rank = os.getenv('LOCAL_RANK', None)
    if __name__ == '__main__' and (local_rank is None or local_rank == 0):
        print("DeepSpeed Adam optimizers are not available. Some features may be unavailable.")
```